tracing-core/src/metadata.rs

```diff
@@ -60,11 +61,11 @@ use crate::stdlib::{
 /// [callsite identifier]: ../callsite/struct.Identifier.html
 pub struct Metadata<'a> {
     /// The name of the span described by this metadata.
-    name: &'static str,
+    name: Cow<'a, str>,
 
     /// The part of the system that the span that this metadata describes
     /// occurred in.
-    target: &'a str,
+    target: Cow<'a, str>,
 
     /// The level of verbosity of the described span.
     level: Level,
```

Comment:
we probably want to use `Cow` `module_path` and `file` as well...

Reply:
`FieldSet` too? (https://github.com/dvdplm/tracing/blob/381d574fd0930c4dabf753c55a769e2ebe454e05/tracing-core/src/field.rs#L73)